omod/src/main/java/org/openmrs/module/owa/web/controller/OwaRestController.java

```diff
@@ -151,77 +157,110 @@
 
 	@RequestMapping(value = "/rest/owa/installapp", method = RequestMethod.POST)
 	@ResponseBody
-	public List<App> install(@RequestBody InstallAppRequestObject urlObject, HttpServletRequest request, HttpServletResponse response) throws IOException {
+	public ResponseEntity<List<App>> install(@RequestBody InstallAppRequestObject urlObject,
+	        HttpServletRequest request, HttpServletResponse response) throws IOException {
+		String message;
+		String fileName = null;
 		List<App> appList = new ArrayList<>();
+		Boolean checkInstall = false;
 
-		String url = urlObject.getUrlValue();
-		String fileName = urlObject.getFileName();
-		if(fileName == null || fileName == ""){
-			throw new  NullPointerException(
-				"File name must be passed for installation"
-			);
-		}
-		URL downloadUrl = null;
-		if (ResourceUtils.isUrl(url)) {
-			downloadUrl = new URL(url);
-		}
 		if (Context.hasPrivilege("Manage OWA")) {
-			String message;
 			HttpSession session = request.getSession();
-			if (!url.isEmpty()) {
+			URL downloadUrl = null;
+			String installUrl = urlObject.getUrlValue();
+
+			if (!installUrl.contains(".zip")) {
+				message = messageSourceService.getMessage("Invalid URL");
```

Comment:
In the message properties file, do we have a message with the code or key as "Invalid URL"? https://github.com/openmrs/openmrs-core/blob/master/api/src/main/resources/messages.properties

Reply:
Secondly, when you have error messages like these displayed to the user, it would be great not to use technical terms like URL. Thirdly, the error message should specify the expected file format such that the end user gets to know how to proceed. Does this make sense?

Reply:
Have added the terms to message.properties in the OWA module. I have also updated the messages to be more descriptive